pkg/disk/partition_table.go

```diff
@@ -750,3 +750,20 @@ func (pt *PartitionTable) GetBuildPackages() []string {
 
 	return packages
 }
+
+// GetMountpointSize takes a mountpoint and returns the size of the entity this
+// mountpoint belongs to.
+func (pt *PartitionTable) GetMountpointSize(mountpoint string) (uint64, error) {
+	path := entityPath(pt, mountpoint)
+	if path == nil {
+		return 0, fmt.Errorf("cannot find mountpoint %s", mountpoint)
+	}
+
+	for _, ent := range path {
+		if sizeable, ok := ent.(Sizeable); ok {
+			return sizeable.GetSize(), nil
+		}
+	}
+
+	panic(fmt.Sprintf("no sizeable of the entity path for mountpoint %s", mountpoint))
```

Comment:
I wonder if we could return an internal error here? I guess I'm just not a fan of panics :)

Reply:
In general, panic should be used for cases which are programming errors. This is IMO such case, so it should be acceptable. Using an error would most probably have the same end result, but the idea behind using panic is to not continue the execution when we hit an undefined state.

Reply:
I agree here about that this is the indented usage of panics and fine to use this way. I also don't want to argue this specific case (I have not looked deep into this).

I do think we are a bit too liberal with panics. When I compare how many panics are used in e.g. the go standard library to images I see there is a difference:
```
mvogt@top:/usr/share/go/src$ grep -r panic\( io/ os/ net/ text/ |grep -v _test.go|wc -l
169
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Go                             583          16642          24314         127429

mvogt@top:~/devel/osbuild/images$ git grep panic\(|grep -v _test.go| wc -l
1549
mvogt@top:~/devel/osbuild/images$ cloc ./pkg/ ./internal/
...
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Go                             481           7686          11931          52681
```

For me the difference is that in go a panic aborts the program so when writing code that is reused something that panics a lot is harder to (re)use. And because (unlike python) there is generally no `recover` around codeblock the panic will bring a (potentially many many go routines spawning) program down entirely.

Reply:
And just to be clear, totally fine to leave as is, just wanted to raise awareness and maybe a tiny discussion/re-evaluation of the pattern :)

Reply:
Yeah, this is a clear programming error, and I think that it's fine to `panic` on these - there's something seriously wrong with the partition table, thus I think the safest approach is to get the whole goroutine down, so the errorneous state doesn't spread. I've added a note about the fact that this is a programming error.